### PR TITLE
Parse and convert enums from query string parameters

### PR DESCRIPTION
**Pull request by zaquestion**
- Added integration tests

### CODE REVIEW

File: gengokit/httptransport/httptransport.go [hunk 1]
```diff
@@ -107,11 +107,18 @@ func NewBinding(i int, meth *svcdef.ServiceMethod) *Binding {
...
 		}
 
+		// IsEnum needed for ConvertFunc and TypeConversion logic just below
+		newField.IsEnum = field.Type.Enum != nil
 		newField.ConvertFunc, newField.ConvertFuncNeedsErrorCheck = createDecodeConvertFunc(newField)
 		newField.TypeConversion = createDecodeTypeConversion(newField)
 
 		nBinding.Fields = append(nBinding.Fields, &newField)
 
+		// Enums are allowed in query/path parameters, skip warning
```
Comment (adamryman) [inline gengokit/httptransport/httptransport.go:117]: If Enums are allowed as both query and path parameters, we should have a test for the path parameters.

I see a test for use as query parameter, though not one for the path parameter.

We don't need to support it in the path currently, though unless we have a test we should warn about it.

File: gengokit/httptransport/httptransport.go [hunk 2]
```diff
@@ -241,25 +247,31 @@ req.{{.CamelName}} = {{.TypeConversion}}
 func createDecodeConvertFunc(f Field) (string, bool) {
 	needsErrorCheck := true
 	fType := ""
-	switch {
-	case strings.Contains(f.GoType, "uint32"):
+	switch f.GoType {
```
Comment (adamryman) [inline gengokit/httptransport/httptransport.go:262]: Very nice cleanup :+1:

**Review by adamryman — COMMENTED** [review]
Looks good! Though we need a test for Enums in the path if we are no longer warning about it. 

Looks like the query is working though!

File: gengokit/httptransport/httptransport.go [hunk 3]
```diff
@@ -319,16 +343,15 @@ func createDecodeTypeConversion(f Field) string {
...
-	switch {
-	case strings.Contains(f.GoType, "uint32"):
-		fType = "uint32(%s)"
-	case strings.Contains(f.GoType, "int32"):
-		fType = "int32(%s)"
-	case strings.Contains(f.GoType, "float32"):
-		fType = "float32(%s)"
+	switch f.GoType {
+	case "uint32", "int32", "float32":
+		fType = f.GoType + "(%s)"
```
Comment (hasLeland) [inline gengokit/httptransport/httptransport.go:348]: :arrow_up: :exclamation: :tada:

**Review by hasLeland — APPROVED** [review]

**Pull request merged**